src/ngraph/frontend/onnx_import/op/split.cpp

```diff
@@ -18,114 +18,38 @@
-                            : Error{name,
-                                    "tensor cannot be split into " + std::to_string(parts) +
-                                        " equal parts, along axis of length " +
-                                        std::to_string(axis_length)}
-                        {
-                        }
-                    };
-
-                    struct Sum : detail::Error
-                    {
-                        explicit Sum(const std::string& name, std::size_t parts, std::size_t axis)
-                            : Error{name,
-                                    "provided lengths of split parts does not sum up to "
-                                    "length of axis we split on: " +
-                                        std::to_string(parts) + " != " + std::to_string(axis)}
-                        {
-                        }
-                    };
-
-                } // namespace split
-
-            } // namespace op
-
-        } // namespace error
-
         namespace op
         {
             namespace set_1
             {
                 NodeVector split(const Node& node)
                 {
-                    std::shared_ptr<ngraph::Node> input = node.get_ng_inputs().at(0);
-                    auto input_shape = input->get_shape();
-                    std::size_t count_outputs{node.get_output_names().size()};
-                    int64_t axis{node.get_attribute_value<int64_t>("axis", 0)};
-                    std::size_t axis_to_split{static_cast<std::size_t>(axis)};
-                    if (axis < 0)
-                    {
-                        axis_to_split = input_shape.size() + axis;
-                    }
-                    else if (axis_to_split >= input_shape.size())
-                    {
-                        throw error::op::split::OutOfRange{node.get_name()};
-                    }
-                    std::size_t length_axis_to_split{input_shape.at(axis_to_split)};
-                    std::vector<std::size_t> length_parts;
+                    const std::shared_ptr<ngraph::Node> input = node.get_ng_inputs().at(0);
+                    const Shape input_shape = input->get_shape();
+                    const std::size_t count_outputs{node.get_output_names().size()};
+                    const int64_t axis{node.get_attribute_value<int64_t>("axis", 0)};
+
                     try
                     {
-                        length_parts = node.get_attribute_value<std::vector<std::size_t>>("split");
+                        const auto length_parts =
+                            node.get_attribute_value<std::vector<std::size_t>>("split");
+                        const auto fused_split =
+                            std::make_shared<ngraph::op::Split>(input, axis, length_parts);
+                        return fused_split->decompose_op();
                     }
                     catch (const std::exception&)
```

Comment:
Please add some comment explaining the logic behind the reason why we're catching here exceptions.